src/algebra/bounds.lean

```diff
@@ -83,3 +83,33 @@ image2_subset_iff.2 $ λ x hx y hy, mul_mem_upper_bounds_mul hx hy
 (hs.mul ht).mono (subset_lower_bounds_mul s t)
 
 end mul_add
+
+section conditionally_complete_lattice
+
+section right
+
+variables {ι G : Type*} [group G] [conditionally_complete_lattice G]
+  [covariant_class G G (function.swap (*)) (≤)] [nonempty ι] {f : ι → G}
+
+@[to_additive csupr_add] lemma csupr_mul' (hf : bdd_above (set.range f)) (a : G) :
+  (⨆ i, f i) * a = ⨆ i, f i * a :=
+(order_iso.mul_right a).map_csupr hf
+
+@[to_additive csupr_sub] lemma csupr_div' (hf : bdd_above (set.range f)) (a : G) :
+  (⨆ i, f i) / a = ⨆ i, f i / a :=
+by simp only [div_eq_mul_inv, csupr_mul' hf]
```

Comment:
Is there a reason these names are primed? I can't find the unprimed versions

Reply:
I would expect that unprimed versions deal with ordered fields or (semi)rings but I can't find a way to say "a linear ordered ring that is a conditionally complete lattice" with the current setup. I can rename these lemmas to `csupr_mul` and `csupr_div`, if you want.

Reply:
I'd be inclined to remove the `'` for now. If we do end up adding a version for fields, then we probably want a version of `order_iso.mul_right` for fields too, and will have to consider renaming that to be primed too.